plugins/module_utils/foreman_helper.py

```diff
@@ -1778,20 +1778,26 @@ def build_fqn(name, parent=None):
 
 # Helper for puppetclasses
 def ensure_puppetclasses(module, entity_type, entity, expected_puppetclasses=None):
-    puppetclasses_resource = '{0}_classes'.format(entity_type)
     if expected_puppetclasses:
+        puppetclasses_resource = '{0}_classes'.format(entity_type)
         expected_puppetclasses = module.find_puppetclasses(expected_puppetclasses, environment=entity['environment_id'], thin=True)
-    current_puppetclasses = entity.pop('puppetclass_ids', [])
```

Comment:
This is probably, why `puppetclass_ids` was not reported anymore.

Reply:
Yes, but also nothing *updates* this after `ensure_puppeclasses` is done…